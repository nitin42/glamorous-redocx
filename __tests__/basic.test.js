import React from 'react';
import { render, mount } from 'enzyme';
import serializer from 'jest-glamor-react';
import renderer from 'react-test-renderer';
import glamorous from '../src/index';

expect.addSnapshotSerializer(serializer)

test('sanity test', () => {
  const TextComponent = glamorous.Text({color: 'red'})

  const tree = renderer.create(
    <TextComponent>Hello World</TextComponent>
  ).toJSON()

  expect(tree).toMatchSnapshot()
});

test('should not throw any error when called glamorous.primitive()', () => {
  glamorous.Text({})
})

test('should attach a displayName', () => {
  const TextComponent = glamorous.Text({})
  const ImageComponent = glamorous.Image({})
  const BulletComponent = glamorous.BulletItem({})
  const NumberComponent = glamorous.NumberItem({})

  expect(TextComponent.displayName).toBe('glamorous(TEXT)')
  expect(ImageComponent.displayName).toBe('glamorous(IMAGE)')
  expect(BulletComponent.displayName).toBe('glamorous(BULLETITEM)')
  expect(NumberComponent.displayName).toBe('glamorous(NUMBERITEM)')
})

test('should render a component with theme properties', () => {
  const Component = glamorous.Text(
    {
      color: 'red',
    },
    (props, theme) => ({fontSize: theme.size}),
  )
  expect(
    render(<Component theme={{size: 20}} />),
  ).toMatchSnapshot()
})

test('should pass an updated theme when theme prop changes', () => {
  const Component = glamorous.Text(
    {
      color: 'red',
    },
    (props, theme) => ({fontSize: theme.size}),
  )
  const wrapper = mount(<Component theme={{size: 10}} />)
  expect(wrapper).toMatchSnapshot(
    `with theme prop of size 10`,
  )
  wrapper.setProps({theme: {padding: 20}})
  expect(wrapper).toMatchSnapshot(
    `with theme prop of size 20`,
  )
})

test('should render the <Image /> component', () => {
  const ImageComponent = glamorous.Image({width: 100})
  expect(
    <ImageComponent src="./redocx.png" />,
  ).toMatchSnapshot()
})

test('supports props as style properties', () => {
  const tree = renderer.create(
    <glamorous.TEXT color="pink">Hello World</glamorous.TEXT>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
