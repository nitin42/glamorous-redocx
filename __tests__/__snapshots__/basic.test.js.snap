// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sanity test 1`] = `
<TEXT
  style={
    Object {
      "color": "red",
    }
  }
>
  Hello World
</TEXT>
`;

exports[`should render a component with theme properties 1`] = `
<text
  style="color:red;font-size:20px;"
/>
`;

exports[`should render the <Image /> component 1`] = `
<glamorous(IMAGE)
  src="./redocx.png"
/>
`;

exports[`supports props as style properties 1`] = `
<TEXT
  color="pink"
  style={
    Object {
      "color": "pink",
    }
  }
>
  Hello World
</TEXT>
`;

exports[`with theme prop of size 10 1`] = `
<glamorous(TEXT)
  theme={
    Object {
      "size": 10,
    }
  }
>
  <TEXT
    style={
      Object {
        "color": "red",
        "fontSize": 10,
      }
    }
    theme={
      Object {
        "size": 10,
      }
    }
  />
</glamorous(TEXT)>
`;

exports[`with theme prop of size 20 1`] = `
<glamorous(TEXT)
  theme={
    Object {
      "padding": 20,
    }
  }
>
  <TEXT
    style={
      Object {
        "color": "red",
        "fontSize": undefined,
      }
    }
    theme={
      Object {
        "padding": 20,
      }
    }
  />
</glamorous(TEXT)>
`;
